src/aml_type/aml.profile.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const constants = require('../yoti_common/constants');
+const GIVEN_NAMES_ATTR = constants.GIVEN_NAMES_ATTR;
+const FAMILY_NAME_ATTR = constants.FAMILY_NAME_ATTR;
+const SSN_ATTR         = constants.SSN_ATTR;
+const ADDRESS_ATTR     = constants.ADDRESS_ATTR;
+
+exports.AmlProfile = class AmlProfile {
+
+  constructor (givenNames, familyName, amlAddress, ssn = '') {
+    this.givenNames = givenNames;
+    this.familyName = familyName;
+    this.amlAddress = amlAddress;
+    this.ssn = ssn;
+
+    AmlProfile.checkAmlAddress(amlAddress);
+  }
+
+  /**
+   * @param givenNames
+   */
+  setGivenNames (givenNames) {
+    this.givenNames = givenNames;
+  }
+
+  /**
+   * @returns {*}
+   */
+  getGivenNames () {
+    return this.givenNames;
+  }
+
+  /**
+   * @param familyName
+   */
+  setFamilyName (familyName) {
+    this.familyName = familyName;
+  }
+
+  /**
+   * @returns familyName
+   */
+  getFamilyName () {
+    return this.familyName;
+  }
+
+  /**
+   * @param amlAddress
+   */
+  setAmlAddress (amlAddress) {
+    AmlProfile.checkAmlAddress(amlAddress);
+    this.amlAddress = amlAddress;
+  }
+
+  /**
+   * @returns amlAddress
+   */
+  getAmlAddress () {
+    return this.amlAddress;
+  }
+
+  /**
+   * @param ssn
+   */
+  setSsn (ssn) {
+    this.ssn = ssn;
+  }
+
+  /**
+   * @returns ssn
+   */
+  getSsn () {
+    return this.ssn;
+  }
+
+  /**
+   * Check if amlAddress is an object.
+   *
+   * @param amlAddress
+   */
+  static checkAmlAddress (amlAddress) {
```

Comment:
Does using `validate` instead of `check` make more sense?

Reply:
either way is fine by me.

Reply:
I prefer `validate`, `check` might just be checking for its presence